packages/components/src/Button/icon.ts

```diff
@@ -64,10 +64,37 @@ export const iconMargins = (props: ButtonProps) => {
   }
 }
 
+export const iconSizes = (props: ButtonProps) => {
+  let iconSize = 16
+  switch (props.size) {
+    case 'xsmall':
+      iconSize = 12
+      break
+    case 'small':
+      iconSize = 14
+      break
+    case 'large':
+      iconSize = 18
+      break
+    default:
+      iconSize = 16
+  }
+
+  if (props.iconBefore || props.iconAfter) {
+    return css`
+      width: ${iconSize}px;
+      height: ${iconSize}px;
+    `
+  } else {
+    return false
+  }
```

Comment:
Is there a reason to not just _always_ set this? If it's non-conditional than the fewer classes need to be generated within the SC stack so it's worth considering the simplification from that perspective as well as just easier to understand.

```suggestion
  return css`
    width: ${iconSize}px;
    height: ${iconSize}px;
  `
```

Reply:
@lukelooker without the conditional it always returns this css for every button. I tested this and it blew up 30 or so snapshots from anywhere a button was used. which scared me a bit. Testing them visually the snapshot failures were just class name changes and everything visually looked fine, but I did not go through each failure.

I think it will be fine since I believe its only generating a selector that targets an icon in the button, but I will defer to your judgment here.

Reply:
Arg - silly snapshots. I think we should go for the pain of reviewing the snapshot changes now since I don't want someone down the road to think this logic check is required.